moveit_simple/include/moveit_simple/moveit_simple.h

```diff
@@ -145,9 +144,11 @@ class Robot
    * @brief execute a given trajectory
    * @param traj_name - name of trajectory to be executed (must be filled with
    * prior calls to "addTrajPoint".
-   * @return - true if desired trajectory was executed.
+   * \throws <std::runtime_error> (Execution failure)
```

Comment:
Are you suggesting, we programatically parse the error string to determine how we should respond?  The `ex.what()` is useful for logging and display, but it's not generally a good idea to use it for recovering errors.  You left that part out of your code snippet, but try and stub it out.  It's complex and fragile at best.